sdk/test-utils/recorder-v2/src/index.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as url from "url";
+import { DefaultHttpClient, WebResource, WebResourceLike } from "@azure/core-http";
+
+export class RecordingHttpClient extends DefaultHttpClient {
```

Comment:
Is the idea that we'll use this HTTP client instead of the default one from core?

Reply:
Wouldn’t it be possible to specify proxy options during tests? to the HTTP pipelines of the test client?

Reply:
> Is the idea that we'll use this HTTP client instead of the default one from core?

Yes

> Wouldn’t it be possible to specify proxy options during tests? to the HTTP pipelines of the test client?

@sadasant  By modifying the client, we are changing the URL to our proxy tool and putting the actual URL in the headers so that the proxy tool can make the request for us. I'm not sure how we can achieve that with proxy options.

Reply:
Similar to what I had done for perf tests, I'm following the same pattern as of now.
We'll modify httpClient for core-v1 packages and pipeline/policy for core-v2 packages.